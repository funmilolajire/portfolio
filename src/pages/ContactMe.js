import { GetInTouch } from '../components/ContactMe--GiT';
import { ContactForm } from '../components/ContactMe--Form';

//layout component
import { Layout } from '../components/LayoutComponent';

export const ContactMe = () => (
    <Layout pageMeta={{
        title: 'Contact Me'
    }}>
        <GetInTouch />
        <ContactForm />
    </Layout>
)