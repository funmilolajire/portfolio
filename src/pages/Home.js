import { useEffect, useRef } from 'react';
import { useLocation } from 'react-router-dom';
import { Hero } from '../components/Home--Hero';
import { About } from '../components/Home--About';
import { SkillSet } from '../components/Home--SkillSet';

//layout component
import { Layout } from '../components/LayoutComponent';

export const Home = () => {
    const location = useLocation();
    const aboutMeRef = useRef();

    useEffect(() => {
        if (location.pathname === "/#about-me") {
            aboutMeRef.current.scrollIntoView({ behavior: 'smooth' })
        } else if (location.pathname === "/") {
            window.scrollTo({
                top: 0,
                bottom: -window.innerHeight,
                behavior: 'smooth'
            })
        }
    })
    return (
        <>
            <Layout pageMeta={{
                title: 'Home'
            }}>
                <Hero />
                <About aboutMeRef={aboutMeRef} />
                <SkillSet />
            </Layout>
        </>
    )
}