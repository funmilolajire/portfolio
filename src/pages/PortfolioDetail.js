import { Header } from '../components/Header';
import { CTA } from '../components/CallToAction';
import { Footer } from '../components/Footer';

export const PortfolioDetail = () => {
    return (
        <div className="PortfolioDetail">
            <Header />
            <CTA />
            <Footer />
        </div>
    )
}