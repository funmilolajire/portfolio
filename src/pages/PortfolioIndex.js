import { Header } from '../components/Header';
import { CTA } from '../components/CallToAction';
import { Footer } from '../components/Footer';

export const PortfolioIndex = () => {
    return (
        <div className="PortfolioIndex">
            <Header />
            <CTA />
            <Footer />
        </div>
    )
}