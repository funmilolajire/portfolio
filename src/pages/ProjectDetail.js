import { Layout } from '../components/LayoutComponent';
import { useParams } from 'react-router-dom';

export const ProjectDetail = () => {
    const { project } = useParams();
    const capitalizeTitle = title => title.split("-")
        .map(word => word.charAt(0).toUpperCase() + word.substring(1)).join(" ");

    return (
        <>
            <Layout pageMeta={{
                title: `${capitalizeTitle(project)}`
            }}>

            </Layout>
        </>
    )
}
