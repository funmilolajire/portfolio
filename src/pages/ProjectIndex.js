//layout component
import { Layout } from '../components/LayoutComponent';
//import components
import { ProjectsMainIndex } from '../components/Projects--Index';

export const ProjectIndex = () => (
    <Layout pageMeta={{
        title: 'Projects'
    }}>
        <ProjectsMainIndex />
    </Layout>
)
