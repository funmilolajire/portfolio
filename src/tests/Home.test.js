import { Home } from '../pages/Home';
import { shallow } from 'enzyme';
// import renderer from 'react-test-renderer';
// import { render, screen } from '@testing-library/react';

jest.mock('react-router-dom', () => ({
    __esModule: true,
    useLocation: jest.fn().mockReturnValue({
        pathname: '/another-route',
        search: '',
        hash: '',
    }),
}));

describe('Home Page', () => {
    describe('make snapshot of Home page', () => {
        test('matches the snapshot', () => {
            let tree = shallow(<Home />)
            expect(tree).toMatchSnapshot();
        })
    })
})