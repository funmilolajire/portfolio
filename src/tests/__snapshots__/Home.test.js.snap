// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page make snapshot of Home page matches the snapshot 1`] = `
<Layout
  pageMeta={
    Object {
      "title": "Home",
    }
  }
>
  <Hero />
  <About />
  <SkillSet />
</Layout>
`;
